.github/container/Dockerfile.base

```diff
@@ -1,4 +1,4 @@
-ARG BASE_IMAGE=nvidia/cuda:12.2.0-devel-ubuntu22.04
+ARG BASE_IMAGE
```

Comment:
My two cents:

I prefer leaving the default in as that makes this Dockerfile self-contained. Now, if a user finds this Dockerfile and wants to build the base container themselves, they need to know where to look to inspect the "default value" that we assign to `BASE_IMAGE`. If the concern is so that we don't repeat ourselves, we could leave the default in the Dockerfile, and have the default of `on.workflow_call.inputs.BASE_IMAGE` be empty to signify to use the default, OR, if specified, it will overwrite

Reply:
@nouiz had a similar feeling. I moved the explicit version back into the Dockerfile and removed the variable from Actions. If I figure out how to do this,
> leave the default in the Dockerfile, and have the default of on.workflow_call.inputs.BASE_IMAGE be empty to signify to use the default, OR, if specified, it will overwrite

I'll push an update.